modules/server/src/network/resolvers/fetch.ts

```diff
@@ -0,0 +1,68 @@
+import axios, { AxiosError } from 'axios';
+import { DocumentNode } from 'graphql';
+
+import { NetworkQueryVariables } from '.';
+import { fetchGql } from '../gql';
+import { failure, Result, success } from '../result';
+import { ASTtoString } from '../utils/gql';
+import { CONNECTION_STATUS } from './aggregations';
+
+type NetworkQuery = {
+	url: string;
+	gqlQuery: DocumentNode;
+	queryVariables: NetworkQueryVariables;
+};
+
+/**
+ * Query remote connections and handle network responses
+ *
+ * @param query
+ * @returns
+ */
+export const fetchData = async <SuccessType>(
+	query: NetworkQuery,
+): Promise<Result<SuccessType, typeof CONNECTION_STATUS.ERROR>> => {
+	const { url, gqlQuery, queryVariables } = query;
+
+	console.log(`Fetch data starting for ${url}..`);
+
+	try {
+		const response = await fetchGql({
```

Comment:
thought, not challenge or request: curious to learn how this fetchGql is different from the regular ajax call in the OG Arranger utils, and eager to see what kind of improvements and new features this one would bring if the two can be unified into a single reusable one for the whole server

Reply:
think the other existing one also does that if I'm not mistaken... maybe I'm thinking of one in the UI components though 🤔